src/js/utils/ui.js

```diff
@@ -311,7 +311,8 @@ const eventRegisters = {
                 if (e.pointerType !== 'touch' && 'x' in e) {
                     // elementFromPoint to handle an issue where setPointerCapture is causing a pointerout event
                     const overElement = document.elementFromPoint(e.x, e.y);
-                    if (!el.contains(overElement)) {
+                    const overElementFromClient = 'clientX' in e ? document.elementFromPoint(e.clientX, e.clientY) : null;
+                    if (!el.contains(overElement) && !el.contains(overElementFromClient)) {
                         triggerEvent(ui, OUT, e);
                     }
```

Comment:
I prefer that we avoid calling `elementFromPoint` twice. Each call has to do a lot of dom tree walking under the hood. We can just as easily return early if the first check passes, and skip the second call to `contains` when the event is missing `clientX` to optimize this code path. Feel free to OR `||` these if you only want one return statement:

```suggestion
                    if (el.contains(overElement)) {
                        return;
                    }
                    if ('clientX' in e && el.contains(document.elementFromPoint(e.clientX, e.clientY))) {
                        return;
                    }
                    triggerEvent(ui, OUT, e);
```

Reply:
The alternative is, can we always use `clientX` instead of `x`?